index.js

```diff
@@ -1,68 +1,59 @@
 // import express (after npm install express)
-const express = require('express');
+const express = require('express')
 
 // create new express app and save it as "app"
-const app = express();
+const app = express()
 
-//import package.json
-const pack = require('./package');
+// import package.json
+const packageInfo = require('./package')
 
-//import categories.json
-const all_categories = require('./src/data/categories.json').categories;
-
-//import products.json
-const all_products = require('./src/data/products').products;
+// import dataService.js
+const dataService = require('./dataService.js')
 
 // server configuration
-const PORT = 8080;
+const PORT = 8080
 
 // create a route for the app
 app.get('/', (req, res) => {
-  res.send('Hello World');
-});
-
-function construct_master(){
-	var master_products = all_products;
-	master_products.forEach(
-		(product) => product.categoryName = all_categories.filter(
-		// check if id of category element matches product categoryId
-			each_category => each_category.id === product.categoryId)[0].categoryName);
-	return master_products;
+  res.send('Hello World')
+})
+
+/*
+ * Filter out products of specific category
+ */
+function getProductsByCategory (ctyId) {
+  // Convert the map to an Array, filter out required products and
+  // create object from the filtered array
+  return Object.fromEntries(
+    Array.from(dataService.getCombinedProductMap()).filter(
```

Comment:
`Object.keys()` and `Object.values()` will not reutrn both at a time and thus we'll have to query each returned key/value if we are to use those methods. Therefore we'll query the 2 dimensional array and take only the required elements to form an object